IPG-Automotive-UK/react-ui
# TD-3234 Update `RoadPreview` Component

Closes [TD-3234](https://sce.myjetbrains.com/youtrack/issue/TD-3234)
Contributes [TD-3196](https://sce.myjetbrains.com/youtrack/issue/TD-3196)

## Changes




A brief description of what this pull request solves / introduces.

- Move the `RoadPreview` component to its own
- Implement the `FormatLabel`, `FormatVersionLabel`, `FileLabel`, `DateLabel`, `UserLabel` components.
- Bound the above mentioned components in boxes, so that they truncate and size to their content properly
- Adjusted types to fit the new definitions
- Adjusted the stories to pass the props in the correct way due to the type change
- Adjusted tests to test with the new `data-testid`s

Maybe a short description of what is not included and will come in future work where appropriate.

## Dependencies




## UI/UX




![image](https://github.com/user-attachments/assets/6056e13d-dee3-42b4-98a1-53fec4c49230)


## Testing notes

1. Open storybook
2. Check the `RoadPreview` component
3. Make sure the `RoadPreview` component belongs to the `Preview` group in storybook
4. Make sure the component matches [FIGMA](https://www.figma.com/design/WoHtIQKM6eLlOUuzK1zlpB/VIRTO.TEST?node-id=3639-22237&node-type=instance&t=zBvJEPPmn4tRgOqi-0)
5. Make sure that you can adjust the `createdAt`, `file`, `format`, `formatVersion`, `user` values. Make sure that when these values are long truncation happens and nothing exceeds the parent component (the card).



## Author checklist

Before I request a review:



~- [ ] I have populated the deploy-preview with relevant test data.~
